ignite/metrics/accuracy.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import division
+
+import torch
+
+from ignite.metrics.metric import Metric
+from ignite.exceptions import NotComputableError
+
+
+class Accuracy(Metric):
+    """
+    Calculates the accuracy.
+
+    - `update` must receive output of the form `(y_pred, y)`.
+    - `y_pred` must be in the following shape (batch_size, num_categories, ...) or (batch_size, ...)
+    - `y` must be in the following shape (batch_size, ...)
+    """
+    def reset(self):
+        self._num_correct = 0
+        self._num_examples = 0
+
+    def update(self, output):
+        y_pred, y = output
+
+        if not (y.ndimension() == y_pred.ndimension() or y.ndimension() + 1 == y_pred.ndimension()):
+            raise ValueError("y must have shape of (batch_size, ...) " +
+                             "and y_pred must have shape of (batch_size, num_classes, ...) or " +
+                             "(batch_size, ...).")
+
+        if y.ndimension() > 1 and y.shape[1] == 1:
+            y = y.squeeze(dim=1)
+
+        if y_pred.ndimension() > 1 and y_pred.shape[1] == 1:
+            y_pred = y_pred.squeeze(dim=1)
+
+        y_shape = y.shape
+        y_pred_shape = y_pred.shape
+
+        if y.ndimension() + 1 == y_pred.ndimension():
+            y_pred_shape = (y_pred_shape[0], ) + y_pred_shape[2:]
+
+        if not (y_shape == y_pred_shape):
+            raise ValueError("y and y_pred must have compatible shapes.")
+
+        if y_pred.ndimension() == y.ndimension():
```

Comment:
Please insert a comment here, something like that
```
# Binary case: map into 2 classes categorical
```

Reply:
Will do! This can't be done for multilabel. Discussed further below!

Reply:
Please add the comment
```suggestion
        # Binary case: map into 2 classes categorical
        if y_pred.ndimension() == y.ndimension():
```

Reply:
Incorporated!